        try {
            t.close();
        } catch (Exception e) {
            //no-op
        }
